### ISSUE TITLE
New crash with latest release

### ISSUE THREAD

**Issue by campoy** [src-d/gitbase#564]
This query makes `gitbase` crash:

```sql
SELECT lang, COUNT(*) as n
FROM (
    SELECT LANGUAGE(cf.file_path, f.blob_content) as lang
    FROM   files AS f
           NATURAL JOIN commit_files cf
           NATURAL JOIN refs
    WHERE  ref_name = 'HEAD'
) AS t
WHERE lang is not null
GROUP BY lang
ORDER BY 2 DESC
```

```go
goroutine 722 [running]:
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).find(0xc0001b3a40, 0xc028ff6000, 0x61c4, 0x83c9, 0x61c4, 0x60, 0xc00003d3e0, 0x6, 0x6)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:175 +0x37e
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).findAll(0xc0001b3a40, 0xc028ff6000, 0x61c4, 0x83c9, 0x61c4, 0x2444670, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:226 +0x1b7
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).replaceAll(0xc0001b3a40, 0xc028ff6000, 0x61c4, 0x83c9, 0xc034b2bb60, 0x1, 0x8, 0x1334f30, 0x8, 0x1, ...)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:513 +0x6f
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).ReplaceAll(0xc0001b3a40, 0xc028ff6000, 0x61c4, 0x83c9, 0xc034b2bb60, 0x1, 0x8, 0xc028ff6000, 0x61c4, 0x83c9)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:550 +0x83
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer.skipCommentsAndLiterals(0xc028ff6000, 0x61c4, 0x83c9, 0x32, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer/tokenize.go:190 +0xf3
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer.Tokenize(0xc01e8a6000, 0x83c9, 0xfe00, 0x1, 0xc0187c5988, 0x8159ac)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer/tokenize.go:19 +0x173
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.(*classifier).Classify(0x2419fc0, 0xc01e8a6000, 0x83c9, 0xfe00, 0xc01bb422d0, 0xc028bc6000, 0x83c9, 0xc028bc6000)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/classifier.go:47 +0x25d
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguagesBySpecificClassifier(0xc01e8a6000, 0x83c9, 0xfe00, 0xc0324db140, 0x3, 0x3, 0x14bdde0, 0x2419fc0, 0x0, 0x0, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:424 +0xf9
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguagesByClassifier(0xc029a6ef00, 0x2d, 0xc01e8a6000, 0x83c9, 0xfe00, 0xc0324db140, 0x3, 0x3, 0x0, 0x0, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:414 +0xae
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguages(0xc029a6ef00, 0x2d, 0xc01e8a6000, 0x83c9, 0xfe00, 0x8, 0xc01708fa18, 0xc000b88b20)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:128 +0x158
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguage(0xc029a6ef00, 0x2d, 0xc01e8a6000, 0x83c9, 0xfe00, 0x16c160d64b406500, 0xc000ef8360)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:38 +0x53
github.com/src-d/gitbase/internal/function.(*Language).Eval(0xc000ec4380, 0xc01ce32ec0, 0xc0373ebdd0, 0x9, 0x9, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/internal/function/language.go:152 +0x1c6
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/expression.(*Alias).Eval(0xc000ec43a0, 0xc0128b6000, 0xc0373ebdd0, 0x9, 0x9, 0x9, 0x0, 0x0, 0xc0187c5d80)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/expression/alias.go:27 +0x5f
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.filterRow(0xc0128b6000, 0xc000e97cd0, 0x1, 0x1, 0xc0373ebdd0, 0x9, 0x9, 0xc000dcf6c0, 0xc000172660, 0xc0187c5df8, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/project.go:147 +0xd4
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*iter).Next(0xc0122023e0, 0xc000ecbda0, 0xc000172660, 0xc0187c5e38, 0x4dc65c, 0xc000ecbe00)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/project.go:133 +0xc3
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql.(*spanIter).Next(0xc0065b8420, 0xc0187c5eac, 0x3, 0x2, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/session.go:194 +0x47
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).iterPartition(0xc000db53e0, 0x14bf040, 0xc012200000)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:241 +0x251
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).start.func1(0xc000db53e0, 0xc000de3170, 0x14bf040, 0xc012200000)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:168 +0x3f
created by github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).start
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:167 +0x10d
panic: runtime error: index out of range

goroutine 579 [running]:
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).find(0xc0001b3a40, 0xc02de40000, 0x7475, 0x83c9, 0x7475, 0x1, 0xc0040f4000, 0x6, 0x6)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:183 +0x359
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).findAll(0xc0001b3a40, 0xc02de40000, 0x7475, 0x83c9, 0x7475, 0x2444670, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:226 +0x1b7
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).replaceAll(0xc0001b3a40, 0xc02de40000, 0x7475, 0x83c9, 0xc01708fb70, 0x1, 0x8, 0x1334f30, 0x8, 0x1, ...)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:513 +0x6f
github.com/src-d/gitbase/vendor/github.com/moovweb/rubex.(*Regexp).ReplaceAll(0xc0001b3a40, 0xc02de40000, 0x7475, 0x83c9, 0xc01708fb70, 0x1, 0x8, 0xc02de40000, 0x7475, 0x83c9)
	/go/src/github.com/src-d/gitbase/vendor/github.com/moovweb/rubex/regex.go:550 +0x83
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer.skipCommentsAndLiterals(0xc02de40000, 0x7475, 0x83c9, 0x32, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer/tokenize.go:190 +0xf3
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer.Tokenize(0xc036dd6000, 0x83c9, 0xfe00, 0xb, 0xc024e81988, 0x8159ac)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/internal/tokenizer/tokenize.go:19 +0x173
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.(*classifier).Classify(0x2419fc0, 0xc036dd6000, 0x83c9, 0xfe00, 0xc0324db170, 0xc02a772000, 0x83c9, 0xc02a772000)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/classifier.go:47 +0x25d
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguagesBySpecificClassifier(0xc036dd6000, 0x83c9, 0xfe00, 0xc033cadce0, 0x3, 0x3, 0x14bdde0, 0x2419fc0, 0x0, 0x0, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:424 +0xf9
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguagesByClassifier(0xc0189091d0, 0x2d, 0xc036dd6000, 0x83c9, 0xfe00, 0xc033cadce0, 0x3, 0x3, 0x0, 0x0, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:414 +0xae
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguages(0xc0189091d0, 0x2d, 0xc036dd6000, 0x83c9, 0xfe00, 0x8, 0xc0356ca280, 0xc000b88b20)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:128 +0x158
github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry%2ev1.GetLanguage(0xc0189091d0, 0x2d, 0xc036dd6000, 0x83c9, 0xfe00, 0x16c160d64b406500, 0xc002954d40)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/enry.v1/common.go:38 +0x53
github.com/src-d/gitbase/internal/function.(*Language).Eval(0xc000ec4380, 0xc017cb6e00, 0xc0397eac60, 0x9, 0x9, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/internal/function/language.go:152 +0x1c6
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/expression.(*Alias).Eval(0xc000ec43a0, 0xc000ebf900, 0xc0397eac60, 0x9, 0x9, 0x9, 0x0, 0x0, 0xc024e81d80)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/expression/alias.go:27 +0x5f
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.filterRow(0xc000ebf900, 0xc000e97cd0, 0x1, 0x1, 0xc0397eac60, 0x9, 0x9, 0xc000dcf6c0, 0xc000172660, 0xc024e81df8, ...)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/project.go:147 +0xd4
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*iter).Next(0xc002808aa0, 0xc000ecbda0, 0xc000172660, 0xc024e81e38, 0x4dc65c, 0xc000ecbe00)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/project.go:133 +0xc3
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql.(*spanIter).Next(0xc000f60bd0, 0xc024e81eac, 0x3, 0x2, 0x0, 0x0)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/session.go:194 +0x47
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).iterPartition(0xc000db53e0, 0x14bf040, 0xc0010bc040)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:241 +0x251
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).start.func1(0xc000db53e0, 0xc000de3170, 0x14bf040, 0xc0010bc040)
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:168 +0x3f
created by github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan.(*exchangeRowIter).start
	/go/src/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/plan/exchange.go:167 +0x10d
```

This is with the binary from the [latest release](https://github.com/src-d/gitbase/releases/tag/v0.17.0-rc.5).

**Comment by smola** [comment]
@campoy Please, use a build without oniguruma.

**Comment by smola** [comment]
Duplicate of: https://github.com/src-d/gitbase/issues/544

**Comment by campoy** [comment]
This is the binary I got from the release. Are you telling me the release binary is not valid?

**Comment by smola** [comment]
@campoy AFAIK only the `static` version. It was recently introduced in the development cycle for 0.17.0, but it will not be part of the stable release (https://github.com/src-d/gitbase/issues/545).

This is a concurrency issue with oniguruma/rubex (https://github.com/src-d/gitbase/issues/544). So far it's only reproducible in a machine with 96 cores. So, yes, one of the binaries distributed in a _release candidate_ has a bug.

Unless something changes, our plan is as follows:
- Stop distributing binaries compiled with oniguruma (0.17.0)
- Clarify README.md and add a warning about enabling oniguruma (0.17.0)
- Move to a maintained library (maybe onigmo/gonigmo) and try to fix the concurrency issues (0.18.0)

At the moment, the only advice we can offer is to use non-static binaries from the release, or to build from source without the oniguruma tag.
